apps/kcapi-hasher.c

```diff
@@ -706,7 +706,7 @@ static int fipscheck_self(const struct hash_params *params_bin,
 	/* Integrity check of our application. */
 	if (mode == SELFCHECK_CHECK || mode == SELFCHECK_PRINT_SELF) {
 		memset(selfname, 0, sizeof(selfname));
-		selfnamesize = readlink("/proc/self/exe", selfname, BUFSIZE);
+		selfnamesize = readlink("/proc/self/exe", selfname, BUFSIZE - 1);
```

Comment:
I just realized the case when readlink() fills the whole buffer is already covered by the check below (and the rest by the memset() call). Please drop this patch when applying. It seems to be a false positive by Coverity (it should check if we check the return value).